contracts/GenArt721CoreV3.sol

```diff
@@ -300,27 +302,31 @@ contract GenArt721CoreV3 is ERC721, Ownable, IGenArt721CoreContractV3 {
     }
 
     /**
-     * @notice Sets the hash for a given token ID `_tokenId`.
+     * @notice Sets the hash seed for a given token ID `_tokenId`.
      * May only be called by the current randomizer contract.
      * May only be called for tokens that have not already been assigned a
      * non-zero hash.
      * @param _tokenId Token ID to set the hash for.
-     * @param _hash Hash to set for the token ID.
+     * @param _hash Hash seed set for the token ID. Only last 12 bytes will be
+     * used.
      * @dev gas-optimized function name because called during mint sequence
      */
     function setTokenHash_8PT(uint256 _tokenId, bytes32 _hash)
         external
         onlyValidTokenId(_tokenId)
     {
+        OwnerAndHashSeed storage ownerAndHashSeed = _ownersAndHashSeeds[
+            _tokenId
+        ];
         require(
             msg.sender == address(randomizerContract),
             "Only randomizer may set"
         );
         require(
-            tokenIdToHash[_tokenId] == bytes32(0),
-            "Token hash already set."
+            ownerAndHashSeed.hashSeed == bytes12(0),
+            "Token hash already set"
         );
-        tokenIdToHash[_tokenId] = _hash;
+        ownerAndHashSeed.hashSeed = bytes12(_hash);
```

Comment:
I think we may want to output an event here that contains the output of `tokenIdToHash(_hash)`, so that we can more readily index the token hash itself, unless we do so somewhere else that I'm overlooking?

cc @yoshiwarab @lyaunzbe

Reply:
I did consider this, but opted to not emit the event from our core contract via the following logic:
- When using a randomizer that assigns token hash during the mint tx, we can key our subgraph off of the `Mint` event to populate token hash on that block
- If we were to change to an async randomizer in the future, the randomizer could emit a `TokenHashPopulatedAsync` event from the future minter, giving our subgraph something else to key off of.

The benefit of this solution is that we don't have to pay the extra gas during the mint tx for an event. There isn't much of a downside, other than some minor subgraph complexity if we upgrade in the future (although, would be a very minor amount of complexity). I suppose some could feel strongly that the event should be something emitted by the core contract instead of the randomizer. I don't have a strong preference for where events are emitted from though, they are really only intended for off-chain indexing.

Reply:
This rationale tracks to me–so long as we're comfortable with the complexity on the subgraph side needed for launching this V3 contract with our infra+frontend integration. Sounds like you're comfortable with this, which I think is totally fair/reasonable, so SGTM to move forward without added event on Core 👍